### PR TITLE
`azurerm_managed_disk` - add state upgrader from V0 to V1 to fix ID issue

### LINKED ISSUE #18865: Error while upgrading the azurerm provider version: parsing segment "staticMicrosoftCompute": expected the segment "microsoft.compute" to be "Microsoft.Compute"
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.2

### AzureRM Provider Version

3.27.0

### Affected Resource(s)/Data Source(s)

azurerm_managed_disk

### Terraform Configuration Files

```hcl
resource "azurerm_managed_disk" "disk_managed_disk" {
  name                 = "disk-01"
  location             = "norwayeast"
  resource_group_name  = azurerm_resource_group.rg.name
  storage_account_type = "Premium_LRS"
  create_option        = "Empty"
  disk_size_gb         = 50
  tags                 = local.default_tags
}
```

### Debug Output/Panic Output

```shell
Error: parsing "/subscriptions/***/resourceGroups/rg/providers/microsoft.compute/disks/disk-01": parsing segment "staticMicrosoftCompute": expected the segment "microsoft.compute" to be "Microsoft.Compute"
```

### Expected Behaviour

This error shouldn't have been there since there are no changes in the configuration and we're just upgrading the provider version.

### Actual Behaviour

While upgrading the azurerm provider version for existing configuration / deployment with an existing state terraform apply or plan fails with the error: parsing segment "staticMicrosoftCompute": expected the segment "microsoft.compute" to be "Microsoft.Compute"

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

Similar to: https://github.com/hashicorp/terraform-provider-azurerm/issues/18235

### PR DESCRIPTION
**Pull request by myc2h6o**
Fix #18865
Previously, `azurerm_managed_disk` could be imported with `Microsoft.compute` segment, which is fixed after we switch to the new SDK [here](https://github.com/hashicorp/go-azure-sdk/blob/dfbb97a902596d5ad114d7a9a2fea113e257c5f7/resource-manager/compute/2022-03-02/disks/id_disk.go#L110).
Adding a state migration function to migrate old `azurerm_managed_disk` with `Microsoft.compute` to `Microsoft.Compute`.

### CODE REVIEW

File: internal/services/compute/migration/managed_disk_v0_to_v1.go [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+			ValidateFunc: validation.StringInSlice([]string{
+				string(disks.DiskStorageAccountTypesStandardLRS),
+				string(disks.DiskStorageAccountTypesStandardSSDZRS),
+				string(disks.DiskStorageAccountTypesPremiumLRS),
+				string(disks.DiskStorageAccountTypesPremiumVTwoLRS),
+				string(disks.DiskStorageAccountTypesPremiumZRS),
+				string(disks.DiskStorageAccountTypesStandardSSDLRS),
+				string(disks.DiskStorageAccountTypesUltraSSDLRS),
+			}, false),
+			DiffSuppressFunc: suppress.CaseDifference,
```
Comment (stephybun) [inline internal/services/compute/migration/managed_disk_v0_to_v1.go:57]: State migrations only require the Type and Required/Optional/Computed for most fields, things like List of Strings also require the Elem Type, so these fields like these can be removed from the `Schema()`

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/compute/migration/managed_disk_v0_to_v1.go (hunk 1 above)
Comment (myc2h6o) [reply internal/services/compute/migration/managed_disk_v0_to_v1.go]: I see, thanks for the information! I've removed the meta-arguments other than Required/Optional/Computed.

**Review by stephybun — COMMENTED** [review]
@myc2h6o were you able to reproduce this bug? I tried testing the migration path from 2.99 to 3.27 but I didn't get the error reported by the user.

**Comment by myc2h6o** [comment]
@stephybun  yes I'm able to repro with the import in v2.99.0
terraform import azurerm_managed_disk.test subscriptions/***/resourceGroups/test/providers/Microsoft.**compute**/disks/test

In [older version](https://github.com/hashicorp/terraform-provider-azurerm/blob/57b41ea7ea305c956d476387d2d197aa2231bfb5/internal/services/compute/managed_disk_resource.go#L809), the parse didn't check for casing of `Microsoft.Compute`, so it is possible to import a disk with the above ID with lower-cased `compute`

**Review by stephybun — APPROVED** [review]
Thanks @myc2h6o LGTM 👍

**Pull request merged**